### ISSUE TITLE
Different logits for the very first generation (llamacpp_hf)

### ISSUE THREAD

**Issue by BadisG** [oobabooga/text-generation-webui#5434]
Hello,

I noticed something interesting, when you start the very first generation of a prompt (meaning the model is doing the prompt processing calculation), you'll get a different logit compared to the second generation and onwards (when you regenerate), here's an example:

```
### Instruction:
When Was born Cédric Doumbé?

### Response:
Cédric Doumbé was born on
```
![cwZHQrb](https://github.com/oobabooga/text-generation-webui/assets/110173477/70cf910e-cdbf-49c2-9e78-21da8a25f5c3)

He was born in "August", and you notice that you're more likely to get the right answer in the very first generation than in the second and subsequent ones. It's just an anecdotical example but I also felt that I got more "luck" in the very first generation compared to the times I had to regenerate, and that logits disparity show that it wasn't just a feeling, it's real.

The main issue there is that if you increase the number of context (like in a chat) you'll always be in the "2nd generation phase" because the model isn't doing the prompt processing mechanism anymore.

I feel it has something to do with this issue https://github.com/oobabooga/text-generation-webui/issues/5133 where the n_batch value isn't impacting the outputs anymore, I think that the very first generation is the one with n_batch = 2048 and the 2nd one and onwards revert back to n_batch = 512, it's just a baseless theory of mine though.

Edit: This bug doesn't appear on llamacpp, only the _hf seems to have this issue. So it means there's an issue with the ```generate_reply_HF``` function compared to the ```generate_reply_custom``` function in ```text_generation.py```

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

1. Take a prompt, and go to "Logits", hit "Get next token probabilities"
2. Hit "Get next token probabilities" again

### Logs

```shell
No logs
```

### System Info

```shell
Windows 10
```
